src/libjson/jsonprint/main.c

```diff
@@ -0,0 +1,125 @@
+/*
+**==============================================================================
+**
+** Copyright (c) Microsoft Corporation
+**
+** All rights reserved.
+**
+** MIT License
+**
+** Permission is hereby granted, free of charge, to any person obtaining a copy
+** of this software and associated documentation files (the ""Software""), to
+** deal in the Software without restriction, including without limitation the
+** rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+** sell copies of the Software, and to permit persons to whom the Software is
+** furnished to do so, subject to the following conditions: The above copyright
+** notice and this permission notice shall be included in all copies or
+** substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+** THE SOFTWARE.
+**
+**==============================================================================
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <json.h>
+#include <sys/stat.h>
+#include <string.h>
+
+static void _write(void* stream, const void* buf, size_t count)
+{
+    fwrite(buf, 1, count, (FILE*)stream);
+}
+
+int _load_file(
```

Comment:
This function is duplicated in several places, can we put it in something like `libjson/util.c`. sgx-lkl could benefit from it too if we expose the API.

Reply:
I moved it to libjson/common and it is shared by the samples. But I cannot put it in **libjson** because it has no dependencies on a C runtime and so no way to load a file.